lib/linter.js

```diff
@@ -94,6 +94,10 @@ function parseJsonConfig(string, location) {
 
     // Parses a JSON-like comment by the same way as parsing CLI option.
     try {
+
+        // https://github.com/eslint/eslint/issues/9366
+        string = string.replace(/\\\\/g, "\\");
```

Comment:
@not-an-aardvark  it can fix the issue. but with a side effect:
```js
// `\` is optional, so either is fine. 
/* eslint id-match: [2, "^(([^$\\W]|\\$[a-f\\d]{2})+|[$_]\\w*|[^\\W\\d]\\w*|[A-Z]([A-Z_]*[A-Z])?)$"*/
/* eslint id-match: [2, "^(([^$\W]|\$[a-f\d]{2})+|[$_]\w*|[^\W\d]\w*|[A-Z]([A-Z_]*[A-Z])?)$"*/
```